src/state/transactions/hooks.tsx

```diff
@@ -113,9 +113,24 @@ export function useHasPendingApproval(token?: Token, spender?: string): boolean
           return false
         } else {
           if (tx.info.type !== TransactionType.APPROVAL) return false
-          return tx.info.spender === spender && tx.info.tokenAddress === token.address && isTransactionRecent(tx)
+          const isPendingApproveOrRevoke =
+            tx.info.spender === spender && tx.info.tokenAddress === token.address && isTransactionRecent(tx)
```

Comment:
Return early to simplify subsequent logic:

```suggestion
          if (!(tx.info.spender === spender && tx.info.tokenAddress === token.address && isTransactionRecent(tx))) return false
```